lottie-compose/src/main/java/com/airbnb/lottie/compose/LottieAnimation.kt

```diff
@@ -10,142 +11,226 @@ import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
+        LaunchedEffect(context, composition, imageAssetsFolder, imageAssetDelegate) {
+            imageAssetManager = ImageAssetManager(context, imageAssetsFolder, imageAssetDelegate, composition.images)
         }
     } else {
         imageAssetManager = null
     }
 
-    SideEffect {
-        drawable.composition = composition
-    }
-
-    // TODO: handle min/max frame setting
-
-    LaunchedEffect(composition, state.isPlaying) {
-        if (!state.isPlaying || composition == null) return@LaunchedEffect
-        var repeatCount = 0
-        if (state.isPlaying && state.progress == 1f) state.progress = 0f
-        var lastFrameTime = withFrameNanos { it }
-        while (true) {
-            withFrameNanos { frameTime ->
-                val dTime = (frameTime - lastFrameTime) / TimeUnit.MILLISECONDS.toNanos(1).toFloat()
-                lastFrameTime = frameTime
-                val dProgress = (dTime * state.speed) / composition.duration
-                val previousProgress = state.progress
-                state.progress = (state.progress + dProgress) % 1f
-                if (previousProgress > state.progress) {
-                    repeatCount++
-                    if (repeatCount != 0 && repeatCount > state.repeatCount) {
-                        state.progress = 1f
-                        state.isPlaying = false
-                    }
-                }
-                val frame = floor(lerp(drawable.minFrame, drawable.maxFrame, state.progress)).toInt()
-                state.updateFrame(frame)
-            }
-        }
-    }
-
-    if (composition == null || composition.duration == 0f) return
-
     Canvas(
         modifier = modifier
             .maintainAspectRatio(composition)
     ) {
         drawIntoCanvas { canvas ->
-            state.applyTo(drawable)
-            drawable.setImageAssetManager(imageAssetManager)
             withTransform({
                 scale(size.width / composition.bounds.width().toFloat(), size.height / composition.bounds.height().toFloat(), Offset.Zero)
             }) {
+                drawable.composition = composition
+                drawable.setOutlineMasksAndMattes(outlineMasksAndMattes)
+                drawable.isApplyingOpacityToLayersEnabled = applyOpacityToLayers
+                drawable.enableMergePathsForKitKatAndAbove(enableMergePaths)
+                drawable.setImageAssetManager(imageAssetManager)
+                drawable.progress = progress
                 drawable.draw(canvas.nativeCanvas)
             }
         }
     }
 }
 
+/**
+ * This is like [LottieAnimation] except that it takes a [LottieCompositionSpec] instead of the
+ * [LottieComposition] itself. This will be more convenient for most use cases unless you
+ * need to explicitly check for failure states while loading the composition, which is done
+ * via [lottieComposition].
+ *
+ * @see LottieAnimation
+ */
+@Composable
+fun LottieAnimation(
```

Comment:
This composable could use a different name, since it's entirely driven by a progress instead of owning the animation.

Reply:
@doris4lt any suggestions on the name?

Reply:
ProgressBasedLottie? ProgressDrivenLottie? LottieByProgress? :D Something that implies: 1) it doesn't animate on its own and 2) driven by a progress would add clarity.

Reply:
Naming is hard :smile: All Lottie animations are driven by progress though. The difference is whether it's internal or external. I think I can remove the overload that takes the spec directly which would reduce this to two overloads. Do you think that would sufficiently help?

Reply:
I removed the extra overloads so there are only 2 now and I think it makes sense for them to have the same name. I honestly can't think of a better name here 😄